### PR TITLE
OIG- remove hard coded listen addresses for servers

### PR DESCRIPTION
**Pull request by rishiagarwal-oracle**
Signed-off-by: Rishi Agarwal <rishi.agarwal@oracle.com>

remove hard coded listen addresses for servers

### CODE REVIEW

File: OracleIdentityGovernance/samples/containerizedDB/docker-compose.yaml [hunk 1]
```diff
@@ -48,6 +52,11 @@ services:
       - DB_PASSWORD=${DC_ORCL_SYSPWD}
       - DB_SCHEMA_PASSWORD=${DC_RCU_SCHPWD}
       - RCUPREFIX=${DC_RCU_OIMPFX}
+    healthcheck:
+      test: ["CMD", "curl", "-fL", "http://oimadmin:7001/console"]
```
Comment (Djelibeybi) [inline OracleIdentityGovernance/samples/containerizedDB/docker-compose.yaml:56]: If you're allowing the hostname to change, wouldn't this test need to change as well?

File: OracleIdentityGovernance/samples/containerizedDB/docker-compose.yaml [hunk 2]
```diff
@@ -68,14 +77,17 @@ services:
...
       - ADMIN_HOST=${DC_HOSTNAME}
       - ADMIN_PORT=7001
-      - adminhostname=${DC_HOSTNAME}
-      - adminport=7001
+      - ADMIN_PASSWORD=${DC_ADMIN_PWD}
       - MANAGED_SERVER=soa_server1
-      - server=soa_server1
+      - MS_HOST=${DC_HOSTNAME}
+    healthcheck:
+      test: ["CMD", "curl", "-f", "http://soams:8001/soa-infra"]
```
Comment (Djelibeybi) [inline OracleIdentityGovernance/samples/containerizedDB/docker-compose.yaml:84]: If you're allowing the hostname to change, wouldn't this test need to change as well?

File: OracleIdentityGovernance/samples/externalDB/docker-compose.yaml [hunk 3]
```diff
@@ -32,6 +32,11 @@ services:
       - DB_PASSWORD=${DC_ORCL_SYSPWD}
       - DB_SCHEMA_PASSWORD=${DC_RCU_SCHPWD}
       - RCUPREFIX=${DC_RCU_OIMPFX}
+    healthcheck:
+      test: ["CMD", "curl", "-fL", "http://oimadmin:7001/console"]
```
Comment (Djelibeybi) [inline OracleIdentityGovernance/samples/externalDB/docker-compose.yaml:36]: If you're allowing the hostname to change, wouldn't this test need to change as well?

File: OracleIdentityGovernance/samples/externalDB/docker-compose.yaml [hunk 4]
```diff
@@ -52,14 +57,17 @@ services:
...
       - ADMIN_HOST=${DC_HOSTNAME}
       - ADMIN_PORT=7001
-      - adminhostname=${DC_HOSTNAME}
-      - adminport=7001
+      - ADMIN_PASSWORD=${DC_ADMIN_PWD}
       - MANAGED_SERVER=soa_server1
-      - server=soa_server1
+      - MS_HOST=${DC_HOSTNAME}
+    healthcheck:
+      test: ["CMD", "curl", "-f", "http://soams:8001/soa-infra"]
```
Comment (Djelibeybi) [inline OracleIdentityGovernance/samples/externalDB/docker-compose.yaml:64]: If you're allowing the hostname to change, wouldn't this test need to change as well?

File: OracleIdentityGovernance/samples/externalDB/docker-compose.yaml [hunk 5]
```diff
@@ -78,9 +86,13 @@ services:
       - ADMIN_HOST=${DC_HOSTNAME}
       - ADMIN_PORT=7001
       - ADMIN_PASSWORD=${DC_ADMIN_PWD}
-      - adminhostname=${DC_HOSTNAME}
-      - adminport=7001
       - MANAGED_SERVER=oim_server1
-      - server=oim_server1
+      - MS_HOST=${DC_HOSTNAME}
+    healthcheck:
+      test: ["CMD", "curl", "-f", "http://oimms:14000/identity"]
```
Comment (Djelibeybi) [inline OracleIdentityGovernance/samples/externalDB/docker-compose.yaml:92]: If you're allowing the hostname to change, wouldn't this test need to change as well?

**Review by Djelibeybi — COMMENTED** [review]
If you're allowing the hostname to change, wouldn't the hostname need to change for the health check?

File: OracleIdentityGovernance/samples/containerizedDB/docker-compose.yaml (hunk 1 above)
Comment (rishiagarwal-oracle) [reply OracleIdentityGovernance/samples/containerizedDB/docker-compose.yaml:56]: Actually the PR description was a bit misleading.
Earlier the listen addresses were being set to hard coded values (oimadmin/oimms/soams) in container scripts
These hard coded values have been removed and IPs are used. 
We are not allowing the hostname to change. 
Hardcoded values in container scripts were preventing spinning up of containers using other methods like in k8s where the host values were different from these hard coded values

Comment (Djelibeybi) [reply OracleIdentityGovernance/samples/containerizedDB/docker-compose.yaml:56]: But these are using hostnames. Wouldn't these health checks fail in the same circumstances?

File: OracleIdentityGovernance/dockerfiles/12.2.1.4.0/container-scripts/createOIMDomain.py [hunk 6]
```diff
@@ -291,16 +291,16 @@ def extendOimDomain(self, domainHome, db, dbPrefix, dbPassword, user, password,
 
         cd('/')
         cd('/Server/AdminServer')
-        cmo.setListenAddress('oimadmin')
+        #cmo.setListenAddress('oimadmin')
```
Comment (Djelibeybi) [inline OracleIdentityGovernance/dockerfiles/12.2.1.4.0/container-scripts/createOIMDomain.py:294]: Please do not just comment out lines. If you don't need the line, remove it. This should apply to all commented lines.

File: OracleIdentityGovernance/samples/containerizedDB/docker-compose.yaml (hunk 1 above)
Comment (rishiagarwal-oracle) [reply OracleIdentityGovernance/samples/containerizedDB/docker-compose.yaml:56]: This health check is specific to the docker-compose. As in the compose, the hostnames are fixed, it works well

Comment (Djelibeybi) [reply OracleIdentityGovernance/samples/containerizedDB/docker-compose.yaml:56]: Ok, great. Just wanted confirmation.

**Review by Djelibeybi — APPROVED** [review]

**Pull request merged**